sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -437,6 +437,23 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
   private def putStateIntoStateCacheMap(
       newVersion: Long,
       map: HDFSBackedStateStoreMap): Unit = synchronized {
+    val loadedEntries = loadedMaps.size()
+    val earliestLoadedVersion: Option[Long] = if (loadedEntries > 0) {
+      Some(loadedMaps.lastKey())
+    } else {
+      None
+    }
+
+    if (earliestLoadedVersion.isDefined) {
+      logInfo(s"Trying to add version=$newVersion to state cache map with " +
+        s"current_size=$loadedEntries and earliest_loaded_version=${earliestLoadedVersion.get} " +
```

Comment:
nit: Would it be sufficient for only knowing earliest loaded version, or do we want to know the "range"?

Reply:
I think its ok to just know earliest ? we know current size anyway and versions are supposed to sequential here since we call this function on each commit ?

Reply:
Yeah probably could calculate back. Since we are not required to look into this message every time (probably only to investigate) so some effort on calculating back is probably OK.